pyomo/core/base/plugin.py

```diff
@@ -220,16 +208,23 @@ def __getitem__(self, name):
         return self._data[name]
 
 
-class Transformation(Plugin):
+class Transformation(object):
     """
     Base class for all model transformations.
     """
-
-    implements(IModelTransformation, service=False)
-
     def __init__(self, **kwds):
         kwds["name"] = kwds.get("name", "transformation")
-        super(Transformation, self).__init__(**kwds)
+        #super(Transformation, self).__init__(**kwds)
+
+    #
+    # Support "with" statements. Forgetting to call deactivate
+    # on Plugins is a common source of memory leaks
+    #
+    def __enter__(self):
+        return self
+
+    def __exit__(self, t, v, traceback):
+        pass
```

Comment:
I do not see why we are adding context manager support to transformations.  Was that something that was implicitly supported by all plugins?

Reply:
Same answer